src/x11_utils.rs

```diff
@@ -50,6 +50,15 @@ pub trait TryParseFd: Sized {
     ) -> Result<(Self, &'a [u8]), ParseError>;
 }
 
+impl<T: TryParse> TryParseFd for T {
```

Comment:
Random thought: Would it be a simplification to remove the `TryParse` trait? (Afterwards `TryParseFD` could be renamed to `TryParse`)

Downside: Lots of code needs to be changed to add an extra argument that is just ignored (oh and the `use` for `RawFdContainer` also becomes necessary everywhere).
Upside: One less `trait`. Does that count as a simplification?

I tend towards "keep both `TryParse` and `TryParseFD`" since FDs are such a special case. Thus, they shouldn't "infect" everything. Still, I am leaving this in case anyone feels like this is a great idea.

Reply:
I also thought about this and came to a similar conclusion.